core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java

```diff
@@ -441,29 +449,49 @@ public void updateUsedBytes(String tierAlias, long usedBytesOnTier) {
   }
 
   /**
-   * Returns the lock for worker register status.
-   *
-   * @return a mutex lock
+   * Enum for the locks in {@link MasterWorkerInfo}.
    */
-  public ReentrantLock getRegisterLock() {
-    return mRegisterLock;
+  public enum LockType {
```

Comment:
Can you move this into its own class, so clients don't need to type encompassing classes?